packages/vite/src/node/config.ts

```diff
@@ -1,18 +1,13 @@
 import fs from 'fs'
 import path from 'path'
-import { Plugin } from './plugin'
-import { BuildOptions, resolveBuildOptions } from './build'
-import {
-  ResolvedServerOptions,
-  resolveServerOptions,
-  ServerOptions
-} from './server'
-import {
-  ResolvedPreviewOptions,
-  resolvePreviewOptions,
-  PreviewOptions
-} from './preview'
-import { CSSOptions } from './plugins/css'
+import type { Plugin } from './plugin'
+import type { BuildOptions } from './build'
+import { resolveBuildOptions } from './build'
+import type { ResolvedServerOptions, ServerOptions } from './server'
+import { resolveServerOptions } from './server'
+import type { ResolvedPreviewOptions, PreviewOptions } from './preview'
+import { resolvePreviewOptions } from './preview'
+import type { CSSOptions } from './plugins/css'
```

Comment:
What's the benefit of forcing this tho?

To me, it was mainly to avoid accidentally introducing side-effects from the modules that you only need the types from. But if you are already imported some runtime from a module, the `type` specifier kinda lost the advantage.

I guess the new syntax in 4.5 will be better, but currently, it breaks many imports from one line to two, which kinda makes the code harder to read and maintain to me.

Reply:
Yes you are right that the TS 4.5 syntax could help, BUT I would like to use eslint fixer for that

The two lines are sometimes better cause they are then not wrapped cause of `printWidth`
Also the imports at the top are mostly not for read but auto organized

When TS 4.5 syntax is supported by @typescript-eslint we can switch to the option

Reply:
Tracking https://github.com/typescript-eslint/typescript-eslint/issues/3950

Reply:
Tracking https://github.com/typescript-eslint/typescript-eslint/issues/2769